app/controllers/experiments_controller.rb

```diff
@@ -237,49 +239,25 @@ def move_modal
 
   # POST: move_experiment(id)
   def move
-    project = Project.find_by_id(params[:experiment].try(:[], :project_id))
-    old_project = @experiment.project
-
-    # Try to move the experiment
-    success = true
-    if @experiment.moveable_projects(current_user).include?(project)
-      success = @experiment.move_to_project(project)
-    else
-      success = false
-    end
+    project = Project.find params[:experiment][:project_id]
 
-    if success
-      Activity.create(
-        type_of: :move_experiment,
-        project: project,
-        experiment: @experiment,
-        user: current_user,
-        message: I18n.t(
-          'activities.move_experiment',
-          user: current_user.full_name,
-          experiment: @experiment.name,
-          project_new: project.name,
-          project_original: old_project.name
-        )
-      )
+    service = Experiments::MoveToProjectService
+              .call(experiment_id: @experiment.id,
+                    project_id: project.id,
+                    user_id: current_user.id)
```

Comment:
The IDs are, I believe, used because the code is executed in delayed job, and therefore serialized, and it's better practice to store IDs than {{ActiveRecord}} objects. Correct me if I'm wrong @urbanrotnik :slightly_smiling_face:.

Reply:
Yeap, as @Ducz0r wrote. We had problems with `sidekiq` jobs for example.

Reply:
@urbanrotnik, @Ducz0r What exactly does it make it execute in the delayed job? Is there missing `.delay` call then on the service? or `handle_async` directive somewhere?

Not sure how well this works with user workflow, because he has to wait on modal for it to happen. Is this expected?

Reply:
I just tested it out without worker running and it's working alright, therefore I assume it's synchronous.

Reply:
@jbargu 
You are right. This works synchronously, actually `generate_image_workflow`  executed with `delay job`.  
Hm.. so here is the question... 
Should we go with a consistent service call, key word arguments, without objects for all services and stay independent to (a)synchronous execution or implementation could vary from service to service?

Reply:
I do belive we have 2 different meanings of services on the table, correct me if I'm wrong. 

a) @urbanrotnik you have in mind an OS type of service (doing something in the background, such as for example `NetworkManager.service` or `dbus.service` on linux - in case of `systemd` or dunno Windows Firewall on Windows, which all run all the time in the background.
b) RoR type of service: offloading some code (usually from the model to make it slimmer) that performs a specific single action. Basically an equivalent of a utility class in OOP doing exactly one thing. Async/sync doesn't really matter here.

So to answer, I believe this behaviour here should be synchronous in the current implementation. 

a) User click on the button to move experiment
b) backend logic performs its magic, 
c) User gets a message: "move successful/unsuccessful"

Moving this logic into async service would mean a heap of troubles. What happens in between when user clicks move, and the experiment is still waiting to be moved, but then someone else comes and changes something on the same experiment? Too many race conditions for my taste.

In case of the sync/async services then as you mentioned @urbanrotnik , one could just create the `AsyncService` class extending the current `Service` class and call 'delay' on the `.call` method.

Please correct me if there's something wrong with my reasoning.

Reply:
* My bad - you are correct, this is not executed in the delayed worker.
* As far as I know, @urbanrotnik is mostly speaking about RoR services, where our aim is indeed, to isolate the business logic into its own hierarchy, away from fat models.
* Regarding the IDs, I'm really not sold on one thing or another. I don't think we will be able to standardize this across the whole board, and I see a point specifically for async services. I personally don't mind having IDs, but of course object passing here would work as well.
* I also agree with @jbargu that this code can remain (mostly, with the exception of image generation) synchronous.

Reply:
@Ducz0r 
True, I was speaking about RoR services. 
@jbargu
As you wrote Jure. Yeap, an implementation should remain synchronous for now. I was thinking if at one point we would use this `Service` triggered as asynchronous job as you mentioned... But you are right, probably not and at least this is not important now...